src/charm.py

```diff
@@ -1,104 +1,432 @@
+
+APPLICATION_CONNECTOR_TYPE_HTTP = "http"
+APPLICATION_CONNECTOR_TYPE_HTTPS = "https"
+
+VALID_APPLICATION_LOG_LEVEL_SETTINGS = [
+    "INFO", "WARN", "DEBUG", "TRACE", "OFF"]
+
+GITLAB_PROJECT_VISIBILITY_PUBLIC = "public"
+GITLAB_PROJECT_VISIBILITY_PRIVATE = "private"
+GITLAB_REQUIRED_SCOPES = ["openid", "profile", "api"]
+GITLAB_OPENID_DISCOVERY_URL = (
+    "https://gitlab.com/.well-known/openid-configuration")
+
 
-class LegendEngineServerOperatorCharm(CharmBase):
-    """Charm the service."""
+class LegendSDLCServerOperatorCharm(charm.CharmBase):
+    """ Charmed operator for the FINOS Legend SDLC Server. """
 
-    _stored = StoredState()
+    _stored = framework.StoredState()
 
     def __init__(self, *args):
         super().__init__(*args)
-        self.framework.observe(self.on.httpbin_pebble_ready, self._on_httpbin_pebble_ready)
-        self.framework.observe(self.on.config_changed, self._on_config_changed)
-        self.framework.observe(self.on.fortune_action, self._on_fortune_action)
-        self._stored.set_default(things=[])
 
-    def _on_httpbin_pebble_ready(self, event):
-        """Define and start a workload using the Pebble API.
+        self._set_stored_defaults()
 
-        TEMPLATE-TODO: change this example to suit your needs.
-        You'll need to specify the right entrypoint and environment
-        configuration for your specific workload. Tip: you can see the
-        standard entrypoint of an existing container using docker inspect
+        self.ingress = ingress.IngressRequires(
+            self,
+            {
+                "service-hostname": self.app.name,
+                "service-name": self.app.name,
+                "service-port": self.model.config[
+                    'server-application-connector-port-http'],
+            },
+        )
+
+        # Standard charm lifecycle events:
+        self.framework.observe(
+            self.on.config_changed, self._on_config_changed)
+        self.framework.observe(
+            self.on.sdlc_pebble_ready, self._on_sdlc_pebble_ready)
+
+        # DB relation lifecycle events:
+        self.framework.observe(
+            self.on["legend-db"].relation_joined,
+            self._on_db_relation_joined)
+        self.framework.observe(
+            self.on["legend-db"].relation_changed,
+            self._on_db_relation_changed)
 
-        Learn more about Pebble layers at https://github.com/canonical/pebble
+        # Studio relation events:
+        self.framework.observe(
+            self.on["legend-sdlc"].relation_joined,
+            self._on_studio_relation_joined)
+        self.framework.observe(
+            self.on["legend-sdlc"].relation_changed,
+            self._on_studio_relation_changed)
+
+    def _set_stored_defaults(self) -> None:
+        self._stored.set_default(log_level="DEBUG")
+        self._stored.set_default(mongodb_credentials={})
+
+    def _on_sdlc_pebble_ready(self, event: framework.EventBase) -> None:
```

Comment:
```suggestion
    def _on_sdlc_pebble_ready(self, event: PebbleReadyEvent) -> None:
```

will need `from ops.charm import PebbleReadyEvent` too